### DISCUSSION TITLE
The connection to the server localhost:8080 was refused - did you specify the right host or port?

### DISCUSSION THREAD

**Discussion by JWheelMaker** [k3s-io/k3s#6217] [General]
**Environmental Info:**
K3s Version: v1.24.6+k3s1 (a8e0c66d)

Node(s) CPU architecture, OS, and Version: Raspberry Pi 4B arm64 newest Version of Raspbeery Pi OS

Cluster Configuration: 2 Pi 4B in the same network to get into one k3s cluster

**Describe the bug:**
On master node i can see both nodes by executing `kubectl get nodes`. But when I use the command on the second node I receive to following error:

> The connection to the server localhost:8080 was refused - did you specify the right host or port?

**Steps To Reproduce:**

- Installed K3s the usual way and added the second node to the cluster

**Expected behavior:**

**Actual behavior:**

**Additional context / logs:**

**Comment by brandond** [comment]
Did you make the second node a server, or an agent? Agents don't have an admin kubeconfig on them; you're not expected to be able to run kubectl there.

**Reply by JWheelMaker** [reply]
Yes exactly. So it is right that it acts the way it does?

**Reply by brandond** [reply]
Yes. If you want to use kubectl on hosts other than the server, you need to copy the admin kubeconfig from the server.

**Reply by JWheelMaker** [reply]
Oh ok. That sounds less like a problem than I first thought :). How can I apply the admin kubeconfig to the worker? I just installed the k3s-agent so there is no k3s.yaml?

**Reply by brandond** [reply]
However you normally copy files between hosts... scp, copy-paste, floppy disk...

**Reply by JWheelMaker** [reply]
Ok yea that makes sence... xD
Tanks for your help :)
